internal/impl/redpanda/processor_data_transforms_test.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package redpanda
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/redpanda-data/benthos/v4/public/service"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func defaultConfig() dataTransformConfig {
+	var cfg dataTransformConfig
+	cfg.maxMemoryPages = 100
+	cfg.timeout = 10 * time.Second
```

Comment:
Nit: Would it be possible to reduce this timeout? The tests run in under a second on my machine.

Reply:
Done